cmd/keeper/keeper.go

```diff
@@ -278,14 +278,29 @@ func (p *PostgresKeeper) createPGParameters(db *cluster.DB) common.Parameters {
 		for _, synchronousStandby := range db.Spec.SynchronousStandbys {
 			synchronousStandbys = append(synchronousStandbys, common.StolonName(synchronousStandby))
 		}
+
+		var syncStandbyNumber int = len(synchronousStandbys)
+
+		// if AdditionalSyncStandbyNames is not empty then consider it as possible
+		// synchronous standby. It can be the PgBarman for instance.
+		// If two nodes stolon cluster fails then the applications specified in AdditionalSyncStandbyNames
+		// will be engaged for the synchronous replication.
+		// The "syncStandbyNumber" variable should not be affected here because otherwise the logic will require synchronous
+		// commits for all standby names
+		if len(db.Spec.AdditionalSyncStandbyNames) > 0 {
+			for _, syncStandByName := range db.Spec.AdditionalSyncStandbyNames {
+				synchronousStandbys = append(synchronousStandbys, syncStandByName)
+			}
+		}
+
```

Comment:
I honestly can't understand the logic here. As I tried to explain in my previous comment: https://github.com/sorintlab/stolon/pull/410/files#r161692823 just drop this and let the sentinel add the AdditionalSyncStandbyNames to the `db.Spec.SynchronousStandby`.